src/bms.rs

```diff
@@ -99,19 +100,19 @@ pub struct BmsOutput {
 /// println!("Warnings: {:?}", warnings);
 /// ```
 #[cfg(feature = "rand")]
-pub fn parse_bms(source: &str) -> BmsOutput {
+pub fn parse_bms(source: &str) -> BmsOutput<BeatKey> {
     use rand::{SeedableRng, rngs::StdRng};
 
     // Parse BMS using default RNG and prompt handler
     let rng = RandRng(StdRng::from_os_rng());
-    parse_bms_with_rng(source, rng)
+    parse_bms_with_rng::<BeatKey>(source, rng)
 }
 
 /// Parse a BMS file from source text using a custom random number generator.
 ///
 /// This function provides a convenient way to parse a BMS file in one step.
 /// It uses the default channel parser and a custom random number generator.
-pub fn parse_bms_with_rng(source: &str, rng: impl Rng) -> BmsOutput {
+pub fn parse_bms_with_rng<T: PhysicalKey + Default>(source: &str, rng: impl Rng) -> BmsOutput<T> {
```

Comment:
The cause of required `Default` is `Bms` and its children are auto-deriving `Default`. `#[derive(..)]` requires the same trait to its type parameters, so `T` is required `Default`. However, data of `Bms` and its children don't seem required `T: Default`. So you can impl them manually to remove the parameter trait bound `Default`.